unity-environment/Assets/ML-Agents/Scripts/Monitor.cs

```diff
@@ -194,34 +198,25 @@ static void InstantiateCanvas()
             canvas.name = "AgentMonitorCanvas";
             canvas.AddComponent<Monitor>();
         }
-        displayTransformValues = new Dictionary<Transform, Dictionary<string, DisplayValue>>();
+        displayTransformValues = new Dictionary<Transform,
+        Dictionary<string, DisplayValue>>();
     }
 
     /// Convert the input object to a float array. Returns a float[0] if the
     /// conversion process fails.
     float[] ToFloatArray(object input)
     {
-        try
-        {
-            return JsonConvert.DeserializeObject<float[]>(
-                JsonConvert.SerializeObject(input, Formatting.None));
-        }
-        catch
-        {
-
-        }
-        try
+        if (input.GetType() == typeof(List<float>))
         {
-            return new float[1]
-            {JsonConvert.DeserializeObject<float>(
-                    JsonConvert.SerializeObject(input, Formatting.None))
-            };
+            input = (input as List<float>).ToArray();
         }
-        catch
+        System.Type inputType = input.GetType();
```

Comment:
There must be a nicer way to do that. Why is the DisplayValue::value object? Can it be anything other than float/array of floats? Also unit testing this method will save you future grief.

Reply:
the value can be a string, a float or an array of float in the current implementation

Reply:
But if it's a string, you ignore it and return a float array, no?